compiler/src/dotty/tools/dotc/core/TypeErasure.scala

```diff
@@ -654,7 +654,14 @@ class TypeErasure(sourceLanguage: SourceLanguage, semiEraseVCs: Boolean, isConst
               tr1 :: trs1.filterNot(_.isAnyRef)
             case nil => nil
           }
-        val erasedDecls = decls.filteredScope(sym => !sym.isType || sym.isClass)
+        var erasedDecls = decls.filteredScope(sym => !sym.isType || sym.isClass).openForMutations
+        for dcl <- erasedDecls.iterator do
+          if dcl.lastKnownDenotation.unforcedAnnotation(defn.TargetNameAnnot).isDefined
+             && dcl.targetName != dcl.name
+          then
+            if erasedDecls eq decls then erasedDecls = erasedDecls.cloneScope
```

Comment:
Should the `openForMutations` be inserted here instead of immediately on line `var erasedDecls`?
```suggestion
            if erasedDecls eq decls then erasedDecls = erasedDecls.cloneScope.openForMutations
```

Reply:
No because `erasedDecls` static type must be `MutableScope`.